src/VisualStudio/Core/Def/Implementation/ExtractClass/VisualStudioExtractClassOptionsService.cs

```diff
@@ -0,0 +1,127 @@
+using Microsoft.CodeAnalysis.Notification;
+using Microsoft.CodeAnalysis.PullMemberUp;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.LanguageServices.Implementation.PullMemberUp;
+using Microsoft.VisualStudio.LanguageServices.Implementation.PullMemberUp.MainDialog;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ExtractClass
+{
+    [ExportWorkspaceService(typeof(IExtractClassOptionsService), ServiceLayer.Host), Shared]
+    internal class VisualStudioExtractClassOptionsService : IExtractClassOptionsService
+    {
+        private readonly IThreadingContext _threadingContext;
+        private readonly IGlyphService _glyphService;
+        private readonly IWaitIndicator _waitIndicator;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public VisualStudioExtractClassOptionsService(
+            IThreadingContext threadingContext,
+            IGlyphService glyphService,
+            IWaitIndicator waitIndicator)
+        {
+            _threadingContext = threadingContext;
+            _glyphService = glyphService;
+            _waitIndicator = waitIndicator;
+        }
+
+        public async Task<ExtractClassOptions?> GetExtractClassOptionsAsync(Document document, INamedTypeSymbol selectedType, ISymbol? selectedMember)
+        {
+            var notificationService = document.Project.Solution.Workspace.Services.GetRequiredService<INotificationService>();
+
+            var membersInType = selectedType.GetMembers().
+               WhereAsArray(member => MemberAndDestinationValidator.IsMemberValid(member));
+
+            var memberViewModels = membersInType
+                .SelectAsArray(member =>
+                    new PullMemberUpSymbolViewModel(member, _glyphService)
+                    {
+                        // The member user selected will be checked at the beginning.
+                        IsChecked = SymbolEquivalenceComparer.Instance.Equals(selectedMember, member),
+                        MakeAbstract = false,
+                        IsMakeAbstractCheckable = !member.IsKind(SymbolKind.Field) && !member.IsAbstract,
+                        IsCheckable = true
+                    });
+
+            using var cancellationTokenSource = new CancellationTokenSource();
+            var memberToDependentsMap = SymbolDependentsBuilder.FindMemberToDependentsMap(membersInType, document.Project, cancellationTokenSource.Token);
+
+            var conflictingTypeNames = selectedType.ContainingNamespace.GetAllTypes(cancellationTokenSource.Token).Select(t => t.Name);
+            var candidateName = selectedType.Name + "Base";
+            var defaultTypeName = NameGenerator.GenerateUniqueName(candidateName, name => !conflictingTypeNames.Contains(name));
```

Comment:
so this ensure uniqueness in the namespace (good), but demonstrates why i worry about you looking up things by name in teh compilation, as i have no idea how that works with types in namespaces.

Reply:
No longer looking up by name in compilation